ros_ws/src/user_package/user_package/user.py

```diff
@@ -0,0 +1,43 @@
+import rclpy
+from rclpy.node import Node
+from example_interfaces.msg import String
+
+class User_Node(Node):
+    def __init__(self):
+        super().__init__("user")
+        self.get_logger().info("User Started")
+        self.request_publisher_=self.create_publisher(String,'user_request',10)
+        #define subscriptions
+        self.subscriber_=self.create_subscription(String, 'clarifying_question',self.get_question_and_respond, 10)
+        self.subscriber_=self.create_subscription(String, 'target',self.reset_for_next_request, 10)
+
+    def initiate_request(self):
+        request=input('What can I get for you?')
+        self.publish_request_or_answer(request)
+
+    def get_question_and_respond(self, question:String):
+        print(question.data)
+        response=input('?:')
+        #convert to String
+        self.publish_request_or_answer(response)
+    
+    def publish_request_or_answer(self,request:str):
+        request_msg=String()
+        request_msg.data=request
+        self.request_publisher_.publish(request_msg)
+
+    def reset_for_next_request(self, target):
+        self.initiate_request()
+
+
+def main(args=None):
+    rclpy.init(args=args)
+    user=User_Node()
+    user.initiate_request()
```

Comment:
This not the right way to do this, as we never reach `.spin` until after the request is done. The way this is written we will need to kill the whole of the user program between each request. Am I reasoning about this wrong?

While this may need to broadcast your user input, it may not need to be a node; alternatively, if it is a node, we need to reason about parallelism better.

Reply:
I love tot talk this through.  Basicially we have to have a chat with the user, then resolve it and reset everything.  Here I start the users ask.  then when "target" is broadcast, it resets the chat.  so we get the looping, its justs triggered inside the node